doc/modules/multiclass.rst

```diff
@@ -14,52 +14,122 @@ Multiclass and multilabel algorithms
-  takes possible values in a finite set of species such as "pear", "apple"; 
-  and "blue" or "green" for a second output variable that takes possible values
-  in a finite set of colors such as "green", "red", "blue", "yellow"...
-
-  This means that any classifiers handling multi-output
-  multiclass or multi-task classification tasks,
-  support the multi-label classification task as a special case.
-  Multi-task classification is similar to the multi-output
-  classification task with different model formulations. For
-  more information, see the relevant estimator documentation.
+- **Multiclass classification**: classification task with more than two classes.
+  Each sample can only be labelled as one class.
+
+  For example, classification using features extracted from a set of images of
+  fruit, where each image may either be of an orange, an apple, or a pear.
+  Each image is one sample and is labelled as one of the 3 possible classes.
+  Multiclass classification makes the assumption that each sample is assigned
+  to one and only one label - one sample cannot, for example, be both a pear
+  and an apple.
+  
+  Valid :term:`multiclass` representations for
+  :func:`~utils.multiclass.type_of_target` (`y`) are:
+
+    - 1d or column vector containing more than two discrete values.
+    - sparse :term:`binary` matrix of shape ``(n_samples, n_classes)`` with a
+      single element per row, where each column represents one class.
+
+
+- **Multilabel classification**: classification task which labels each sample
+  with a **set** of target labels. This can be thought of as predicting
+  properties of a sample that are not mutually exclusive. Formally, a value of
+  0 or 1 is assigned to each label, for every sample. It is thus comparable
+  to running 'n_labels' binary classification tasks, for example with
+  :class:`sklearn.multiclass.OneVsRestClassifier`. Multilabel classification
+  differs in that only one classification model is trained whereas 'n_labels'
+  classification models would be trained using the 'n_labels' binary
+  classification tasks approach.
+
+  For example, prediction of taste using features extracted from a set of
+  images of fruit, where the possible taste labels are 'sour', 'sweet',
+  'juicy' and 'bitter'. Each image is one sample and is labelled with a
+  set of target labels, where the set can be any number of labels, ranging from
+  all 4 labels to none of the labels. A good 'real-world' example is predicting
+  the topics relevant to a text document or video. The document or video may be
+  about 'religion', 'politics', 'finance' or 'education', or a subset of these
+  topic labels or even all of these topic labels.
+
+  Valid representation :term:`multilabel` `y` is either dense (or sparse)
+  :term:`binary` matrix of shape ``(n_samples, n_classes)`` with multiple
+  active elements per row to denote that the sample belongs to multiple classes.
```

Comment:
active -> nonzero?

that the sample belongs to multiple classes -> which classes the sample belongs to